Src/DependencyCollector/Shared/DependencyTrackingTelemetryModule.cs

```diff
@@ -219,11 +224,26 @@ protected virtual void Dispose(bool disposing)
         /// <summary>
         /// Initialize for framework event source (not supported for Net40).
         /// </summary>
-        private void InitializeForFrameworkEventSource()
+        private void InitializeForDiagnosticAndFrameworkEventSource()
         {
-#if !NET40
-            FrameworkHttpProcessing frameworkHttpProcessing = new FrameworkHttpProcessing(this.telemetryConfiguration, DependencyTableStore.Instance.WebRequestCacheHolder, this.SetComponentCorrelationHttpHeaders, this.ExcludeComponentCorrelationHttpHeadersOnDomains, this.EffectiveProfileQueryEndpoint);
-            this.httpDesktopDiagnosticSourceListener = new HttpDesktopDiagnosticSourceListener(frameworkHttpProcessing);
+#if NET45
+            if (this.EnableDiagnosticSourceInstrumentation)
+            {
+                DesktopDiagnosticSourceHttpProcessing desktopHttpProcessing = new DesktopDiagnosticSourceHttpProcessing(
+                    this.telemetryConfiguration,
+                    DependencyTableStore.Instance.WebRequestConditionalHolder,
+                    this.SetComponentCorrelationHttpHeaders,
+                    this.ExcludeComponentCorrelationHttpHeadersOnDomains,
+                    this.EffectiveProfileQueryEndpoint);
+                this.httpDesktopDiagnosticSourceListener = new HttpDesktopDiagnosticSourceListener(desktopHttpProcessing);
+            }
+
+            FrameworkHttpProcessing frameworkHttpProcessing = new FrameworkHttpProcessing(
```

Comment:
if EnableDiagnosticSourceInstrumentation is true, we enable it. So why do we need to initialize Framework any more? Even if you enable it, its doing to ignore all the call backs as it has check 'DependencyTableStore.Instance.IsDesktopHttpDiagnosticSourceActivated'. This is not desirable. (correct me if i am missing something."